static/gsApp/views/amCheckout/cart.tsx

```diff
@@ -0,0 +1,641 @@
+        </Item>
+      )}
+      <Item data-test-id="summary-item-due-today">
+        <ItemFlex>
+          <DueToday>{t('Due today')}</DueToday>
+          {previewDataLoading ? (
+            <Placeholder height="24px" width={PRICE_PLACEHOLDER_WIDTH} />
+          ) : (
+            <DueTodayPrice>
+              {originalBilledTotal > billedTotal && (
+                <DueTodayAmountBeforeDiscount>
+                  {utils.displayPrice({
+                    cents: originalBilledTotal,
+                  })}{' '}
+                </DueTodayAmountBeforeDiscount>
+              )}
+              <DueTodayAmount>
+                {utils.displayPrice({
+                  cents: billedTotal,
+                })}
+              </DueTodayAmount>
+              <span> USD</span>
+            </DueTodayPrice>
+          )}
+        </ItemFlex>
+      </Item>
+      <StyledButton
+        aria-label={t('Confirm and pay')}
+        priority="primary"
+        onClick={onSubmit}
+        disabled={buttonDisabled || previewDataLoading}
+      >
+        <IconLock locked />
+        {isSubmitting ? t('Checking out...') : t('Confirm and pay')}
+      </StyledButton>
+      <Subtext>
+        {!!effectiveDate &&
+          tct('Your changes will apply on [effectiveDate]. ', {
+            effectiveDate: moment(effectiveDate).format('MMM D, YYYY'),
+          })}
+        {longInterval === 'yearly'
+          ? tct(
+              'Plan renews [longInterval] on [renewalDate]. Any additional usage will continue to be billed monthly.',
+              {
+                longInterval,
+                renewalDate: moment(renewalDate).format('MMM D, YYYY'),
+              }
+            )
+          : tct(
+              'Plan renews [longInterval] on [renewalDate], plus any additional usage[onDemandLimit].',
+              {
+                longInterval,
+                renewalDate: moment(renewalDate).format('MMM D, YYYY'),
+                onDemandLimit: totalOnDemandBudget
+                  ? tct(' (up to [onDemandMaxSpend]/month)', {
+                      onDemandMaxSpend: utils.displayPrice({
+                        cents: totalOnDemandBudget,
+                      }),
+                    })
+                  : '',
+              }
+            )}
+      </Subtext>
+    </SummarySection>
+  );
+}
+
+function Cart({
+  activePlan,
+  api,
+  formData,
+  subscription,
+  organization,
+  referrer,
+  hasCompleteBillingDetails,
+}: CartProps) {
+  const [previewDataLoading, setPreviewDataLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [previewData, setPreviewData] = useState<PreviewData | null>(null);
+  const [stripe, setStripe] = useState<stripe.Stripe>();
+  const [effectiveDate, setEffectiveDate] = useState<moment.Moment | null>(null); // this is only set when the effective date is in the future
+  const [renewalDate, setRenewalDate] = useState<moment.Moment | null>(null);
```

Comment:
lets store these as dates instead of moment objects just because moment is a weird thing that mutates. call `toDate` before saving them. This way any places downstream also would avoid mutating the moment object